airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/objects/impl/ConnectorSpecificationAdapter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.protocol.objects.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.protocol.objects.ConnectorSpecification;
+import io.airbyte.commons.protocol.objects.DestinationSyncMode;
+import io.airbyte.protocol.models.AdvancedAuth;
+import io.airbyte.protocol.models.AuthSpecification;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode
+public class ConnectorSpecificationAdapter implements ConnectorSpecification {
```

Comment:
checking my understanding: is the idea that we would have an adapter for every protocol version?